python/dgl/convert.py

```diff
@@ -564,16 +564,21 @@ def to_hetero(G, ntypes, etypes, ntype_field=NTYPE, etype_field=ETYPE,
     return to_heterogeneous(G, ntypes, etypes, ntype_field=ntype_field,
                             etype_field=etype_field, metagraph=metagraph)
 
-def to_homogeneous(G, ndata=None, edata=None):
+def to_homogeneous(G, ndata=None, edata=None, store_type=True, return_count=False):
```

Comment:
1. Is there any real use cases asking for `return_count`? 
2. What if we provide a `return_count` API for hetero graphs, other than just a param in `to_homogeneous`?

Reply:
> 1. Is there any real use case asking for `return_count`?

It is for this PR. Feeding `return_count` to the forward function of `RelGraphConv` can avoid the cost of sorting the graph. 

> 2. What if we provide a `return_count` API for hetero graphs, other than just a param in `to_homogeneous`?

The count array is useful when the user use a homogeneous graph to represent a heterograph. It can speed up some operations. Right now it only affects `RelGraphConv`, but we will add more later.

Look like I should add some descriptions about using scenarios of this count array.

Reply:
Providing the senarios sounds good. 

Besides, even we'll consider adding a standalone `return_count` for hetero graph, it doesn't conflict with this `to_homogeneous(return_count)` : we can simply replace the internal implementation with that new API, but keep the input param as is.